internal/uvm/network.go

```diff
@@ -72,36 +72,89 @@ func (uvm *UtilityVM) removeNamespaceNICs(ns *namespaceInfo) error {
 }
 
 func (uvm *UtilityVM) addNIC(id guid.GUID, endpoint *hns.HNSEndpoint) error {
+
+	// First a pre-add. This is a guest-only request and is only done on Windows.
+	if uvm.operatingSystem == "windows" {
+
+		preAddRequest := hcsschema.ModifySettingRequest{
```

Comment:
I'm still confused. Why is this nested 2 levels deep. There is a `RequestType` on the `NetworkModifyRequest` itself and the `GuestRequest` itself. Are we sure that's how Windows expects it?

Reply:
This matches `file:podscenario.tests.ps1` in research as of Friday.

```
unction AddNetworkEndpoint {
    param(
        $sys,
        $hnsendpoint
    )

    $resourceID = New-Guid

    # PreAdd
    & {
        $networkRequest = [HCS.Schema.Requests.NetworkModifySettingRequest]::new()
        $networkRequest.AdapterInstanceId = $resourceID
        $networkRequest.RequestType = "PreAdd"
        $networkRequest.Settings = $hnsendpoint

        $guestRequest = [HCS.Schema.Requests.ModifySettingRequest]::new()
        $guestRequest.ResourceType = "Network"
        $guestRequest.RequestType = "Add"
        $guestRequest.Settings = $networkRequest

        $request = [HCS.Schema.Requests.ModifySettingRequest]::new()
        $request.GuestRequest = $guestRequest
        Modify-HcsSystem $sys (ConvertTo-HcsJson $request)
    }

    # Add
    & {
        $networkRequest = [HCS.Schema.Requests.NetworkModifySettingRequest]::new()
        $networkRequest.AdapterInstanceId = $resourceID
        $networkRequest.RequestType = "Add"

        $guestRequest = [HCS.Schema.Requests.ModifySettingRequest]::new()
        $guestRequest.ResourceType = "Network"
        $guestRequest.RequestType = "Add"
        $guestRequest.Settings = $networkRequest

        $adapter = [HCS.Schema.VirtualMachines.Resources.Network.NetworkAdapter]@{
            EndpointId = $hnsendpoint.ID;
            MacAddress = $hnsendpoint.MacAddress;
        }
        $request = [HCS.Schema.Requests.ModifySettingRequest]@{
            ResourcePath = "VirtualMachine/Devices/NetworkAdapters/$resourceID";
            ResourceType = "Network";
            RequestType  = "Add";
            Settings     = $adapter;
            GuestRequest = $guestRequest;
        }

        Modify-HcsSystem $sys (ConvertTo-HcsJson $request)
    }

    return $resourceID
}
```

Reply:
Ok sounds good then